e2e-tests/src/synthetic_network.rs

```diff
@@ -22,3 +33,58 @@ pub async fn set_out_latency(milliseconds: Milliseconds, synthetic_url: String)
         .await
         .expect("unable to commit network configuration");
 }
+
+pub async fn wait_for_further_finalized_blocks(
+    connection: &SignedConnection,
+    blocks_to_wait: u32,
+) -> anyhow::Result<()> {
+    let finalized = connection.get_finalized_block_hash().await?;
+    let finalized_number = connection
+        .get_block_number(finalized)
+        .await?
+        .ok_or(anyhow!(
+            "Failed to retrieve block number for hash {finalized:?}"
+        ))?;
+    let block_number_to_wait = finalized_number + blocks_to_wait;
+    info!(
+        "Current finalized block #{}, waiting for block #{}",
+        finalized_number, block_number_to_wait
+    );
+    connection
+        .wait_for_block(|n| n > block_number_to_wait, BlockStatus::Finalized)
+        .await;
+    Ok(())
+}
+
+pub async fn test_latency_template_test(
+    config: &Config,
+    validator_count: usize,
+    out_latency: Milliseconds,
+) -> anyhow::Result<()> {
+    let connections = config.create_signed_connections().await;
+    join_all(
+        config
+            .synthetic_network_urls()
+            .into_iter()
+            .take(validator_count)
+            .map(|synthetic_url| set_out_latency(out_latency, synthetic_url)),
+    )
+    .await;
+    info!("Waiting for session 1");
+    join_all(
+        connections
+            .iter()
+            .map(|connection| connection.wait_for_session(1, BlockStatus::Finalized)),
+    )
+    .await;
+    let blocks_to_wait_in_first_session = 30;
+    info!(
+        "Waiting for {} finalized blocks in sesssion 1 to make sure initial unit collection works.",
+        blocks_to_wait_in_first_session
+    );
+    join_all(connections.iter().map(|connection| {
```

Comment:
Use `try_join_all` or don't return `Result` and panic.

Reply:
Good point, done